pages/api/lockerboxes/validateqr.ts

```diff
@@ -0,0 +1,42 @@
+import type { NextApiRequest, NextApiResponse } from 'next'
+import prisma from '../../../lib/prisma'
+import * as argon2 from 'argon2'
+import { Status } from '../../../types/types'
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  if (req.method === 'POST') {
+    try {
+      const { qrCode } = req.body
+      let unfilled_prescriptions = await prisma.prescription.findMany({
+        where: {
+          status: Status.Unfilled
+        }
+      })
+
+      unfilled_prescriptions.forEach(async (prescription) => {
+        let is_valid_code = await argon2.verify(qrCode, prescription.pickupCode)
+        if (is_valid_code) {
+          await prisma.prescription.update({
+            data: {
+              status: Status.Filled
+            },
+            where: {
+              id: prescription.id
+            }
+          })
+
+          res.status(200).json({ message: ' Success', qrCode })
+        }
+      })
+
+      res.status(200).json({ message: 'INVALID QR code', qrCode })
```

Comment:
This should be an `if else` because the update is async, but only for that context. That means that this function will return a state with an invalid code if prisma is still updating and will cause a race condition. i.e we can have a response of an invalid code when we have a true code. You can optionally place the entire lookup in a promise but that seems like an over complication
i.e
```ts
if(isValidCode){
// yay valid code
// do some async stuff
res.status(200).json({ ... })
}else{
// invalid code boo!
res.status(400).json({...}}
}
```